Standard sample bracketing

archaeothommy opened the pull request:
@Andreaacme2:  Is the function ready for merging? If not, what are the to dos?

alexandrarodler committed:
Rare earth element data normalization to PAAS and C1

alexandrarodler committed:
Merge branch 'FlowerDataProcessing' of https://github.com/archaeothommy/ASTR into FlowerDataProcessing

Andreaacme2 committed:
New editing

Andreaacme2 committed:
Merge conflict

Merge branch 'FlowerDataProcessing' of github.com:archaeothommy/ASTR into FlowerDataProcessing

# Conflicts:
#	R/Bracketing.R

Andreaacme2 committed:
Resolved merged conflict

alexandrarodler committed:
Publication data normalization

alexandrarodler committed:
Merge branch 'FlowerDataProcessing' of https://github.com/archaeothommy/ASTR into FlowerDataProcessing

alexandrarodler committed:
REY_PAAS-C1-normalization

alexandrarodler committed:
FlowerDataProcessing-SBB-test-03

alexandrarodler committed:
FlowerDataProcessing-SBB-test-04

Andreaacme2 committed:
New editions

ChenWang-Leiden committed:
vignettes of Bracketing function

Andreaacme2 committed:
New new edits for the Bracketing

ChenWang-Leiden committed:
Merge branch 'FlowerDataProcessing' of https://github.com/archaeothommy/ASTR into FlowerDataProcessing

Andreaacme2 committed:
New editing for bracketing software

Andreaacme2 committed:
Merge branch 'FlowerDataProcessing' of github.com:archaeothommy/ASTR into FlowerDataProcessing

alexandrarodler committed:
FlowerDataProcessing-SBB-test-03

Andreaacme2 committed:
Cycle management and weight standards during bracketing

Andreaacme2 committed:
check

Andreaacme2 committed:
checking

Andreaacme2 committed:
Merge branch 'FlowerDataProcessing' of github.com:archaeothommy/ASTR into FlowerDataProcessing

ChenWang-Leiden committed:
Update Bracketing_vignette.Rmd

unify the SSB linear in the Equations.

ChenWang-Leiden committed:
Update Bracketing_vignette.Rmd

uniform

Andreaacme2 committed:
Description of parameters for the bracketing function

Commit:
update formatting and documentation

Commit:
revise SSB function:
* consistent coding style
* remove unused variables and unecessary declarations
* remove implicit assumption that ID column has fixed name
* increase clarity and consistency of input and documentation.
* add additional input checks

Andreaacme2 committed:
Change of "se" and "se_error" variables into "sdev" and "sd_dev", given it represents standard deviation.

Andreaacme2 committed:
-Deletion of cycle_size variable. Now cycles can have different lenghts
-Addition of sd_input variable for displaying 1SD, 2SD, ..., nSD.
-Addition of SBB examples in the data_raw folder

Andreaacme2 committed:
Deletion of test file

Andreaacme2 committed:
-Addition of new variable "notation" to define which calculation the user wants to do.
-Deletion of linear SSB

archaeothommy committed:
lint function, update notation

archaeothommy commented:
@Andreaacme2 Thank you so much for implementing the changes. The function looks awesome now! 🤩 
I changed the possible values for indicating the notation because the previous ones were so similar that they were prone to typos. 

Can I merge the function?

Andreaacme2 commented:
@archaeothommy No worries! and sounds good. You can merge the function now.
Later on I will see which other editings can be done to make the function more flexible.

archaeothommy committed:
merging main

archaeothommy committed:
prepare for merge to main

archaeothommy commented:
@Andreaacme2 : last question, lines 173-177 seems to be duplicates of lines 161-165 in the `while`-loop. Are they a leftover or is this on purpose?

archaeothommy committed:
remove training whitesprace [skip ci]

Andreaacme2 committed:
Changes in the SSB function:
-Addition of id_error parameter
-Change variable id_std for std
-change of variable values for id_values
-Changes in the main structure of the function by using the library dplyr

Commit:
improving SSB function:
* update examples, test and vignette to new function structure
* revise documentation to include default values
* update column names of output to be closer to  the ASTR format
* declare dependencies
* linting

archaeothommy commented:
Handling of the error is tricky... After your work during the CAA workshop, the function now requires a column with the analytical precisions. However, 

* The summary does not take the individual errors into account but calculates the SD based on the SSB corrected values. If intentional, this must be properly documented as one would expect otherwise because the errors must be given but are irrelevant for the summary output. 
* The detailed results report the errors for the invidual sample as intended. However, 
    * The column name is "Error_err2SD" but based on the code, I think they are not in 2SD unless the input happens to be... 
    * Mustn't the error be scaled together with the values when calculating the delta and epsilon notation?
